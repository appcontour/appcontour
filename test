#!/bin/bash
#

l__ist=`ls tests/test.??`
if [ -n "$1" ]
then
  l__ist=tests/test.$1
fi

tmpf__ile=test$$.tmp
failed="0"

for f__ile in $l__ist
do
  i__nfo=$(grep "^#testname:" $f__ile | cut -f2- -d:)
  if [ -n "$i__nfo" ]; then i__nfo=" ($i__nfo)"; fi
  test=`echo $f__ile | cut -f2 -d\.`
  echo "Running test $test$i__nfo..."
  source $f__ile 2>/dev/null >$tmpf__ile
  exitcode=$?

  if [ -f ${f__ile}.out ]
  then
    if ! diff $tmpf__ile ${f__ile}.out
    then
      failed=$[ $failed + 1 ]
    fi
  else
    echo "...new test, copying out file"
    cp $tmpf__ile ${f__ile}.out
  fi
  rm $tmpf__ile
  echo "$exitcode" >$tmpf__ile
  if [ -f ${f__ile}.ec ]
  then
    if ! diff $tmpf__ile ${f__ile}.ec
    then
      failed=$[ $failed + 1 ]
    fi
  else
    echo "...new test, copying ec file"
    cp $tmpf__ile ${f__ile}.ec
  fi
  rm $tmpf__ile
done

if [ "$failed" -gt "0" ]
then
  echo $failed tests failed
  exit 1
fi

exit 0
