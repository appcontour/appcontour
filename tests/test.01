./contour applyrule n1 <examples/tre_sfere.morse
