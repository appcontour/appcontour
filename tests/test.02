#./contour applyrule n1 <examples/three_spheres.morse | ./contour applyrule n1 | ./contour applyrule n1 | ./contour applyrule n1
./contour applyrule k2 <examples/three_spheres.morse | ./contour applyrule k2 | ./contour applyrule k2 | ./contour applyrule k2
