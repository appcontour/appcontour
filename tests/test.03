./contour applyrule n1 <examples/tre_sfere.morse | ./contour applyrule n1 | ./contour applyrule n1
