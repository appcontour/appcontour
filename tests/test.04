./contour ishuffman -q examples/two_rings.morse
./contour iscontour -q examples/immersed/boysurface.morse
