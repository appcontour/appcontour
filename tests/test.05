./contour applyrule n2 <examples/sfera_dietro_toro.morse
./contour applyrule n3 <examples/sfera_davanti_toro.morse
./contour applyrule n4 <examples/due_anelli.morse | ./contour applyrule n2 | ./contour applyrule n3 | ./contour applyrule n1
./contour applyrule n4 <examples/toro2.morse
./contour applyrule N5 <examples/tre_sfere.morse | ./contour applyrule N5:6 | ./contour applyrule N1 | ./contour applyrule N1 | ./contour applyrule N1
