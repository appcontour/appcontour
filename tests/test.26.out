
=====================================
testing example disjoint_spheres.morse
=====================================
d = 1
1;

=====================================
testing example genus_2_linked.morse
=====================================
d = 2
1;

=====================================
testing example genus_2.morse
=====================================
d = 2
1;

=====================================
testing example hill.morse
=====================================
d = 1
1;

=====================================
testing example hollowsphere.morse
=====================================
d = 1
1;
1;

=====================================
testing example internalexternal.morse
=====================================
d = 2
+1-u+u^2;

=====================================
testing example internalknot.morse
=====================================
d = 1
1;

=====================================
testing example mendes.morse
=====================================
d = 3
1;

=====================================
testing example milnor_a.morse
=====================================
d = 1
1;

=====================================
testing example painted_klein_bottle.morse
=====================================
d = 2
1;

=====================================
testing example partial_eclipse.morse
=====================================
d = 1
1;

=====================================
testing example potato.morse
=====================================
d = 1
1;

=====================================
testing example sphere_2.morse
=====================================
d = 1
1;

=====================================
testing example sphere_behind_torus.morse
=====================================
d = 1
1;

=====================================
testing example sphere_infrontof_torus.morse
=====================================
d = 1
1;

=====================================
testing example sphere.morse
=====================================
d = 1
1;

=====================================
testing example test_rule_cn3.morse
=====================================
d = 1
1;

=====================================
testing example three_spheres.morse
=====================================
d = 1
1;

=====================================
testing example torus1.morse
=====================================
d = 1
1;

=====================================
testing example torus2.morse
=====================================
d = 1
1;

=====================================
testing example torus3.morse
=====================================
d = 1
1;

=====================================
testing example torus_eight.morse
=====================================
d = 1
1;

=====================================
testing example torus_twist.morse
=====================================
d = 1
1;

=====================================
testing example trefoil.morse
=====================================
d = 4
1;

=====================================
testing example two_beans.morse
=====================================
d = 1
1;

=====================================
testing example two_hills.morse
=====================================
d = 1
1;

=====================================
testing example twohollowsphere.morse
=====================================
d = 1
1;
1;
1;

=====================================
testing example two_rings.morse
=====================================
d = 2
1;

=====================================
testing example borromeanrings.knot
=====================================
d = 1
Cannot compute Alexander polynomial for groups with rank 3

=====================================
testing example conway.knot
=====================================
d = 1
+1;

=====================================
testing example conway_pinched.knot
=====================================
d = 2
# *** Warning: result can be noncanonical ***
Ideal [
+1-u-2v+2uv+u^2v+uv^2-2u^2v^2+u^3v^2;
(-1+u-u^2+u^3) (u - 1);
(-1+u-u^2+u^3) (v - 1);
(+2-2u+u^2-u^3-uv+2u^2v-u^3v+u^4v) (u - 1);
(+2-2u+u^2-u^3-uv+2u^2v-u^3v+u^4v) (v - 1);
]

=====================================
testing example double_twist.knot
=====================================
d = 1
1;

=====================================
testing example eight.knot
=====================================
d = 1
1;

=====================================
testing example embrace.knot
=====================================
d = 2
+1+u-v;

=====================================
testing example embrace_trefoil.knot
=====================================
d = 2
+1+u-v;

=====================================
testing example granny.knot
=====================================
d = 1
+1-2t+3t^2-2t^3+t^4;

=====================================
testing example kinoshita_terasaka.knot
=====================================
d = 1
+1;

=====================================
testing example linked_rings.knot
=====================================
d = 1
Ideal [
(+1) (u - 1);
(+1) (v - 1);
]

=====================================
testing example linked_rings_twice.knot
=====================================
d = 1
Ideal [
(+1+u) (u - 1);
(+1+u) (v - 1);
]

=====================================
testing example ochiai_mod_to_genus2.knot
=====================================
d = 2
# *** Warning: result can be noncanonical ***
Ideal [
+1-2u+u^2+2uv-u^2v;
(+1-u) (u - 1);
(+1-u) (v - 1);
]

=====================================
testing example ochiai_mod_to_link.knot
=====================================
d = 1
Ideal [
(+1-u+u^2) (u - 1);
(+1-u+u^2) (v - 1);
]

=====================================
testing example ochiai_unknot.knot
=====================================
d = 1
1;

=====================================
testing example one_chain.knot
=====================================
d = 1
1;

=====================================
testing example square.knot
=====================================
d = 1
+1-2t+3t^2-2t^3+t^4;

=====================================
testing example toric_3_2.knot
=====================================
d = 1
+1-t+t^2;

=====================================
testing example trefoilknotfake.knot
=====================================
d = 1
1;

=====================================
testing example trefoilknot.knot
=====================================
d = 1
+1-t+t^2;

=====================================
testing example trinacria.knot
=====================================
d = 3
Cannot compute Alexander polynomial for groups with rank 3

=====================================
testing example two_rings.knot
=====================================
d = 2
1;

=====================================
testing example whitehead.knot
=====================================
d = 1
Ideal [
(+1-u-v+uv) (u - 1);
(+1-u-v+uv) (v - 1);
]

=====================================
testing example knots/knot_0_1.knot
=====================================
d = 1
1;

=====================================
testing example knots/knot_3_1.knot
=====================================
d = 1
+1-t+t^2;

=====================================
testing example knots/knot_4_1.knot
=====================================
d = 1
-1+3t-t^2;

=====================================
testing example knots/knot_5_1.knot
=====================================
d = 1
+1-t+t^2-t^3+t^4;

=====================================
testing example knots/knot_5_2.knot
=====================================
d = 1
+2-3t+2t^2;

=====================================
testing example knots/knot_6_1.knot
=====================================
d = 1
-2+5t-2t^2;

=====================================
testing example knots/knot_6_2.knot
=====================================
d = 1
+1-3t+3t^2-3t^3+t^4;

=====================================
testing example knots/knot_6_3.knot
=====================================
d = 1
+1-3t+5t^2-3t^3+t^4;

=====================================
testing example knots/knot_7_1.knot
=====================================
d = 1
+1-t+t^2-t^3+t^4-t^5+t^6;

=====================================
testing example knots/knot_7_2.knot
=====================================
d = 1
+3-5t+3t^2;

=====================================
testing example knots/knot_7_3.knot
=====================================
d = 1
+2-3t+3t^2-3t^3+2t^4;

=====================================
testing example knots/knot_7_4.knot
=====================================
d = 1
+4-7t+4t^2;

=====================================
testing example knots/knot_7_5.knot
=====================================
d = 1
+2-4t+5t^2-4t^3+2t^4;

=====================================
testing example knots/knot_7_6.knot
=====================================
d = 1
+1-5t+7t^2-5t^3+t^4;

=====================================
testing example knots/knot_7_7.knot
=====================================
d = 1
+1-5t+9t^2-5t^3+t^4;

=====================================
testing example knots/knot_8_17.knot
=====================================
d = 1
+1-4t+8t^2-11t^3+8t^4-4t^5+t^6;

=====================================
testing example knots/knot_8_19.knot
=====================================
d = 1
+1-t+t^3-t^5+t^6;

=====================================
testing example knots/knot_8_1.knot
=====================================
d = 1
-3+7t-3t^2;

=====================================
testing example knots/knot_8_20.knot
=====================================
d = 1
+1-2t+3t^2-2t^3+t^4;

=====================================
testing example knots/knot_8_21.knot
=====================================
d = 1
+1-4t+5t^2-4t^3+t^4;

=====================================
testing example knots/knot_8_2.knot
=====================================
d = 1
+1-3t+3t^2-3t^3+3t^4-3t^5+t^6;

=====================================
testing example knots/knot_8_3.knot
=====================================
d = 1
-4+9t-4t^2;

=====================================
testing example knots/knot_8_5.knot
=====================================
d = 1
+1-3t+4t^2-5t^3+4t^4-3t^5+t^6;

=====================================
testing example knots/knot_10_132.knot
=====================================
d = 1
+1-t+t^2-t^3+t^4;

=====================================
testing example knots/thistlethwaite.knot
=====================================
d = 1
1;

=====================================
testing example knots/kanenobu_0_0.knot
=====================================
d = 1
+1-6t+11t^2-6t^3+t^4;

=====================================
testing example knots/kanenobu_1_1.knot
=====================================
d = 1
+1-3t+5t^2-7t^3+5t^4-3t^5+t^6;

=====================================
testing example knots/kanenobu_2_0.knot
=====================================
d = 1
+1-6t+11t^2-6t^3+t^4;

=====================================
testing example links/link_2_0_1.knot
=====================================
d = 2
1;

=====================================
testing example links/link_2_2_1.knot
=====================================
d = 1
Ideal [
(+1) (u - 1);
(+1) (v - 1);
]

=====================================
testing example links/link_2_4_1.knot
=====================================
d = 1
Ideal [
(+1+u) (u - 1);
(+1+u) (v - 1);
]

=====================================
testing example links/link_2_5_1.knot
=====================================
d = 1
Ideal [
(+1-u-v+uv) (u - 1);
(+1-u-v+uv) (v - 1);
]

=====================================
testing example links/link_2_6_1.knot
=====================================
d = 1
Ideal [
(+1+u+u^2) (u - 1);
(+1+u+u^2) (v - 1);
]

=====================================
testing example links/link_2_6_2.knot
=====================================
d = 1
Ideal [
(+u^2+v-uv+u^2v+v^2) (u - 1);
(+u^2+v-uv+u^2v+v^2) (v - 1);
]

=====================================
testing example links/link_2_6_3.knot
=====================================
d = 1
Ideal [
(-1+2u+2v-uv) (u - 1);
(-1+2u+2v-uv) (v - 1);
]
